### PR TITLE
[1.13] Bring back version checking and forge beta warning

### PR DESCRIPTION
**Pull request by ichttt**
~~Also return an `ArtifactVersion` from `ForgeVersion`, so we can check the minor version to see if forge is in beta instead of the currently unimplemented version checker~~
Requires https://github.com/MinecraftForge/ForgeSPI/pull/1 to compile
Reimplements the update checker, and the main menu forge beta warning.

### CODE REVIEW

**Comment by LexManos** [comment]
I dont think the switch to ArtifactVersion isn't a good one. Esp considering checking vs 0 doesn't mean it's a beta.
Re-hooking in the VersionCheck system would be a better PR. Shouldn't be to difficult. The json side hasn't changed. And it's still just a V1 > V2 comparison.

**Comment by ichttt** [comment]
Ok, updated

File: src/main/java/net/minecraftforge/fml/VersionChecker.java [hunk 1]
```diff
@@ -103,20 +125,179 @@ private CheckResult(Status status, @Nullable ComparableVersion target, @Nullable
 
     public static void startVersionCheck()
     {
+        new Thread("Forge Version Check")
+        {
+            @Override
+            public void run()
+            {
+//                if (!ForgeModContainer.getConfig().get(ForgeModContainer.VERSION_CHECK_CAT, "Global", true).getBoolean())
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/VersionChecker.java]: ForgeConfig exists.

**Comment by ichttt** [comment]
Alright, config is in. Note that I can't do much about the PR bot erroring out. It is because my SPI PR isn't on the maven yet

**Pull request merged**